### PR TITLE
Reduce code duplication here and there

### PR DESCRIPTION
**Pull request by Jalle19**
Fixes some Scrutinizer complaints

### CODE REVIEW

File: src/Search/Query/Joining/AbstractQuery.php [hunk 1]
```diff
@@ -19,4 +20,59 @@
...
+     *
+     * @throws InvalidArgument
+     */
+    protected function assertScoreMode($scoreMode)
+    {
+        $validModes = $this->getValidScoreModes();
+        
+        if (!in_array($scoreMode, $validModes)) {
+            throw new InvalidArgument(sprintf(
+                'HasChild Query `score_mode` must be one of "%s", "%s" given.',
```
Comment (kryysler) [inline src/Search/Query/Joining/AbstractQuery.php]: Incorrect class name in message

Comment (Jalle19) [inline src/Search/Query/Joining/AbstractQuery.php]: Fixed, also fixed the typo in the commit message

Comment (kryysler) [inline src/Search/Query/Joining/AbstractQuery.php]: 👍

**Comment by kryysler** [comment]
just as a general note, the score mode options are changed in ES 5.0 and all joining queries does not support them in the same way, so that needs the be re-done when upgrading support for 5.x.

**Comment by Jalle19** [comment]
@kryysler once https://github.com/digiaonline/lumen-elasticsearch/pull/17 is merged I think we should move the `Traits` directory one level up and add a few more, i.e. `HasField`, `HasFields`, `HasValue` and `HasValues` since these are duplicated here and there now. We could then also rename the `BoostableQuery` to just `HasBoost` for consistency.

**Comment by kryysler** [comment]
Yep, sounds good.

**Pull request merged**